### ISSUE TITLE
interpretTimeGhcAt together with mkDateTime leads to unexpected milliseconds in current time

### ISSUE THREAD

**Issue by L7R7** [tek/polysemy-time#3]
I'm currently building a project that uses polysemy-time and I was very happy to see that `interpretTimeGhcAt` and `mkDateTime` is in there, especially for testing. However, I found that `now` returns a time that has milliseconds /= 0 and is therefore different than the time I provided in the interpreter. For example:

```hs
testTime :: UTCTime
testTime = mkDatetime 1845 12 31 23 59 59

someFunc :: IO ()
someFunc = do
  now <- runM $ interpretTimeGhcAt testTime Time.now
  print now
  print testTime
  print $ now == testTime -- returns False, was expecting True
```
returns 

```
1845-12-31 23:59:59.000008382 UTC
1845-12-31 23:59:59 UTC
False
```

So it's _almost_ the same, which obviously isn't enough for tests.

What do you think? Am I missing something? Did I find a bug in the lib? 

I have a minimal stack project to reproduce it [here](https://github.com/L7R7/haskell-incubator/blob/main/polysemy-time/src/Lib.hs)

**Comment by tek** [comment]
`interpretTimeGhcAt` only sets the start time, if you want to use a fixed time that doesn't change, you need `interpretTimeGhcConstant`!

**Comment by tek** [comment]
(available in 0.1.4.0)

**Comment by L7R7** [comment]
Oh! That's it. Should have found that myself in the docs.

Thank you for the quick response, keep up the good work!
